### ISSUE TITLE
Disable SnapshotDebugger for local development only?

### ISSUE THREAD

**Issue by gorillapower** [Azure/Azure-Functions#1856]
Is it possible to disable SnapshotDebugger for local development only?

We have the following host.json file
```
{
  "version": "2.0",
  "logging": {
      "applicationInsights": {
      "samplingSettings": {
        "isEnabled": true,
        "maxTelemetryItemsPerSecond": 10,
        "excludedTypes": "Request;Exception",
        "includedTypes": "Dependency;Event;PageView;Trace"
      },
      "snapshotConfiguration": {
        "IsEnabled": true,
        "isEnabledInDeveloperMode": false
      }
    }
  },
  "functionTimeout": "00:10:00"
}
```
When we run our functions locally, we find that exceptions trigger the snapshot debugger to run, snapshotuploader64.exe and attempt to upload a snapshot to Azure. 

As a result, the local machine's bandwith is depleted and slows down all cloud dependency requests (cosmos, redis, azure storage, etc) and the whole machine in general. This process is quite is quite hidden and can be very difficult to troubleshoot if you dont know about it (took me a few days to discover why my local machine was grinding to a halt).

By setting the `'IsEnabled': false`, this problem ceases, however as a development team, we want to be able to keep the `host.json` in SourceControl with the setting `'IsEnabled':true` so that our functions in Azure do record snapshots.

Basically, we want to disable snapshots for local development. But the `isEnabledInDeveloperMode: false` doesnt seem to work. It looks like an ApplicationInsights native setting, but I dont see any documentation about this setting.

If im not mistaken, the `host.json` settings cannot be overidden in code using the `IFunctionsConfigurationBuilder`, but that would be a good solution if it could.

**Comment by v-anvari** [comment]
Hi @gorillapower , There are few possibilities that is causing the scenario to not work. Can you verify if instrumentation key is read from the ApplicationInsights.config file. Can you also go through the [troubleshooting document ](https://docs.microsoft.com/en-us/azure/azure-monitor/app/snapshot-debugger-troubleshoot#verify-the-instrumentation-key) to check if you are missing anything. 

To avoid mixing up telemetry from development, test, and production environments, you can create separate Application Insights resources and change their keys, depending on the environment. 
[api-custom-events-metrics#dynamic-ikey](https://docs.microsoft.com/en-us/azure/azure-monitor/app/api-custom-events-metrics#dynamic-ikey)

**Comment by v-anvari** [comment]
Closing this issue as the required information is provided, please feel free to reach back to us in case of any further queries related to this issue

**Comment by gorillapower** [comment]
@v-anvari Sorry for the delay. Our Azure Function application is picking up the `InstrumentationKey` fine from, however the snapshot debugger is starting on our local machines. I guess my question is, is there a way to disable it for local development only?

**Comment by v-anvari** [comment]
Hi @gorillapower , This is not possible directly, but you can create separate resources like mentioned in the above comment.

"To avoid mixing up telemetry from development, test, and production environments, you can create separate Application Insights resources and change their keys, depending on the environment.
["api-custom-events-metrics#dynamic-ikey"](https://docs.microsoft.com/en-us/azure/azure-monitor/app/api-custom-events-metrics#dynamic-ikey)
